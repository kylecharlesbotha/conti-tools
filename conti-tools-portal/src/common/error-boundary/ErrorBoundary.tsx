/* eslint-disable no-console */
import React, { ErrorInfo } from 'react';

//This particular component is intentionally left as a class-based component due to restrictions in React 16
export class ErrorBoundary extends React.Component<
  Record<string, unknown>,
  { hasError: boolean }
> {
  constructor(props: Readonly<Record<string, unknown>>) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error: Error) {
    // Update state so the next render will show the fallback UI.
    console.error(error);
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    // You can also log the error to an error reporting service
    console.warn('Error was not reported to central logging, please implement');
    console.debug(error, errorInfo);
  }

  render() {
    if (this.state.hasError) {
      // You can render any custom fallback UI
      return <h1>Something went wrong.</h1>;
    }

    return this.props.children;
  }
}
