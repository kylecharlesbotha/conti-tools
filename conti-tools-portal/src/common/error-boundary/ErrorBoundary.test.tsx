import React from 'react';
import { render } from '@testing-library/react';
import { ErrorBoundary } from './ErrorBoundary';

const ThrowAnError: React.FunctionComponent<{ haveAnError?: boolean }> = ({
  children,
  haveAnError
}) => {
  if (haveAnError) {
    throw new Error("I'm a component in crisis! (A fake error)");
  }
  return <>{children}</>;
};
jest.mock('src/lib/get-config/get-config.ts');

it('renders the error boundary correctly when an error occurs', () => {
  const tree = render(
    <ErrorBoundary>
      <ThrowAnError haveAnError={true}>Test</ThrowAnError>
    </ErrorBoundary>
  ).baseElement;
  expect(tree).toMatchSnapshot();
});

it('renders the children correctly when no error occurs', () => {
  const tree = render(
    <ErrorBoundary>
      <ThrowAnError>Test</ThrowAnError>
    </ErrorBoundary>
  ).baseElement;
  expect(tree).toMatchSnapshot();
});
