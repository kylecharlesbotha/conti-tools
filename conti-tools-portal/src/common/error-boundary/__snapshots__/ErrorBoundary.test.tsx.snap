// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the children correctly when no error occurs 1`] = `
<body>
  <div>
    Test
  </div>
</body>
`;

exports[`renders the error boundary correctly when an error occurs 1`] = `
<body>
  <div>
    <h1>
      Something went wrong.
    </h1>
  </div>
</body>
`;
